### PR TITLE
feat(dart): add missing clients

### PR DESCRIPTION
**Pull request by aallam**
## 🧭 What and Why

Add missing clients:
  - [x] Abtesting
  - [x] Analytics
  - [x] Ingestion
  - [x] Monitoring
  - [x] Personalization
  - [x] Predict
  - [x] Query Suggestions
  - [x] Recommend

### Changes included:

- [x] Generate API clients
- [x] Generate CTS

### CODE REVIEW

File: config/openapitools.json [hunk 1]
```diff
@@ -187,7 +187,55 @@
         "additionalProperties": {
           "packageVersion": "0.1.2"
         }
+      },
+      "dart-recommend": {
```
[the remark this answers is not in the document]
Comment (aallam) [reply config/openapitools.json:191]: we can yes

File: clients/algoliasearch-client-dart/packages/client_abtesting/lib/src/extension.dart
Comment (shortcuts) [inline clients/algoliasearch-client-dart/packages/client_abtesting/lib/src/extension.dart:1]: what's the point of those files? do we expect something in it or to differ at some point? if not, I'd move them to templates and let the gen handle them

File: playground/java/src/main/java/com/algolia/playground/Scratch.java
Comment (shortcuts) [inline playground/java/src/main/java/com/algolia/playground/Scratch.java:1]: is it expected to be pushed?

File: templates/dart/serialization/json_serializable/class_data.mustache [hunk 2]
```diff
@@ -26,6 +26,12 @@ final class {{{classname}}}{
   {{#description}}
       /// {{{description}}}
   {{/description}}
+  {{#vendorExtensions}}
+  {{#x-is-oneof}}
+     /// One of types: {{#x-oneof-types}}
+     /// - [{{{.}}}]{{/x-oneof-types}}
```
Comment (shortcuts) [inline templates/dart/serialization/json_serializable/class_data.mustache:32]: should it stay a comment?

Comment (shortcuts) [reply templates/dart/serialization/json_serializable/class_data.mustache:32]: ah yes ok it's really a comment in the description

**Review by shortcuts — COMMENTED** [review]
clean!

File: playground/java/src/main/java/com/algolia/playground/Scratch.java
Comment (aallam) [reply playground/java/src/main/java/com/algolia/playground/Scratch.java:1]: Nope :D

File: clients/algoliasearch-client-dart/packages/client_abtesting/lib/src/extension.dart
Comment (aallam) [reply clients/algoliasearch-client-dart/packages/client_abtesting/lib/src/extension.dart:1]: These files are reserved for custom code or helper in order to provide a better DX, and their content is not automatically generated

**Review by shortcuts — APPROVED** [review]
nothing to add! GG!

**Pull request merged**